state/factory/workingset.go

```diff
@@ -195,12 +200,36 @@ func (ws *workingSet) runAction(
 			)
 		}
 		if receipt != nil {
+			isBlobTx := false // TODO: get blob type from action
+			if protocol.MustGetFeatureCtx(ctx).DisableBlobTransaction || !isBlobTx {
+				return receipt, nil
+			}
+			if err = ws.handleBlob(ctx, selp, receipt); err != nil {
+				if e := ws.Revert(si); err != nil {
+					log.L().Panic("failed to revert snapshot", zap.Error(e))
+				}
+				return nil, err
+			}
```

Comment:
logic here seems incorrect
snapshot is taken before running `Handle()`, so if `Revert(si) == nil`, it will still return `receipt, nil`, but receipt is the result of running `Handle()`, which is now reverted

Reply:
removed revert(), just return error when err occurs during handling blobs